packages/cli/src/commands/publish.ts

```diff
@@ -44,9 +45,14 @@ export default class Publish extends Command {
     } else {
       throw new Error('Please provide SUBQL_ACCESS_TOKEN before publish');
     }
-
-    this.log('Uploading SubQuery project to IPFS');
     const cid = await uploadToIpfs(project.manifest, authToken.trim(), flags.ipfs).catch((e) => this.error(e));
-    this.log(`SubQuery Project uploaded to IPFS: ${cid}`);
+    await createIPFSFile(location, cid);
+
+    if (!flags.output) {
+      this.log('Uploading SubQuery project to IPFS');
+      this.log(`SubQuery Project uploaded to IPFS: ${cid}`);
+    } else {
+      this.log(`${cid}`);
```

Comment:
If this log the cid, wonder if it is easy piped to another command? I thought we should return it here, your thought @stwiname ?